packages/jaeger-ui/src/components/DeepDependencies/Header/index.tsx

```diff
@@ -13,70 +13,91 @@
 // limitations under the License.
 
 import * as React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
 
 import HopsSelector from './HopsSelector';
 import NameSelector from './NameSelector';
-import tempOptions from './tmp-data';
-
+import * as jaegerApiActions from '../../../actions/jaeger-api';
 import { EDirection, TDdgDistanceToPathElems } from '../../../model/ddg/types';
+import { ReduxState } from '../../../types/index';
 
 import './index.css';
 
-type TProps = {
-  distanceToPathElems?: TDdgDistanceToPathElems;
-  setDistance: (distance: number, direction: EDirection) => void;
-  visEncoding?: string;
+type TDispatchProps = {
+  fetchServices: () => void;
+  fetchServiceOperations: (service: string) => void;
 };
 
-type TTempState = {
-  service: string | null;
-  operation: string | null;
+type TReduxProps = {
+  operationsForService: Record<string, string[]>;
+  services?: string[] | null;
 };
 
-// istanbul ignore next
-const TMP_OPTIIONS = tempOptions.map(s => (s.match(/^(\S+\s+){1,12}|^\S+$/) || ['MISSING'])[0]);
-
-export default class Header extends React.PureComponent<TProps, TTempState> {
-  state = {
-    service: null,
-    operation: null,
+type TProps = TDispatchProps &
+  TReduxProps & {
+    distanceToPathElems?: TDdgDistanceToPathElems;
+    operation?: string;
+    service?: string;
+    setDistance: (distance: number, direction: EDirection) => void;
+    setOperation: (operation: string) => void;
+    setService: (service: string) => void;
+    visEncoding?: string;
   };
 
-  setService = (service: string) => {
-    // istanbul ignore next
-    if (service !== this.state.service) {
-      this.setState({ service, operation: null });
+export class HeaderImpl extends React.PureComponent<TProps> {
+  constructor(props: TProps) {
+    super(props);
+
+    const { fetchServices, fetchServiceOperations, operationsForService, service, services } = props;
+
+    if (!services) {
+      fetchServices();
     }
-  };
+    if (service && !Reflect.has(operationsForService, service)) {
+      fetchServiceOperations(service);
+    }
+  }
 
-  setOperation = (operation: string) => {
-    // istanbul ignore next
-    this.setState({ operation });
+  setService = (service: string) => {
+    const { fetchServiceOperations, setService, operationsForService } = this.props;
+    if (!Reflect.has(operationsForService, service)) {
+      fetchServiceOperations(service);
```

Comment:
Having the `Header` fetch the services or operations, if they're needed, is inconsistent with `index.tsx` fetching the deps-graph data instead `Graph` taking care of that.

What do you think?

Reply:
Ping...

Reply:
just pushed the updated version